rust/pact_ffi/src/mock_server/handles.rs

```diff
@@ -1809,6 +1809,81 @@ pub extern fn pactffi_with_binary_file(
   }
 }
 
+ffi_fn!{
+  /// Add matching rules to the interaction.
+  ///
+  /// * `interaction` - Interaction handle to set the matching rules for.
+  /// * `rules` - JSON string of the matching rules to add to the interaction.
+  ///
+  /// This function can be called multiple times, in which case the matching
+  /// rules will be merged. The function will return `true` if the rules were
+  /// successfully added, and `false` if an error occurred.
+  ///
+  /// # Safety
+  ///
+  /// The rules parameter must be a valid pointer to a NULL terminated UTF-8
+  /// string.
+  fn pactffi_with_matching_rules(
+    interaction: InteractionHandle,
+    rules: *const c_char
+  ) -> bool {
+    let rules = match convert_cstr("rules", rules) {
+      Some(rules) => rules,
+      None => {
+        error!("with_matching_rules: Rules value is not valid (NULL or non-UTF-8)");
+        return Ok(false);
+      }
+    };
+
+    let rules = match serde_json::from_str::<Value>(rules) {
+      Ok(Value::Object(rules)) => rules,
+      Ok(_) => {
+        error!("with_matching_rules: Rules value is not a JSON object");
+        return Ok(false);
+      },
+      Err(err) => {
+        error!("with_matching_rules: Failed to parse the matching rules: {}", err);
+        return Ok(false);
+      }
+    };
+
+    // Wrap the rules in a object with a "matchingRules" key if it is not
+    // already, as this is required for `matchers_from_json`.
+    let rules = if rules.contains_key("matchingRules") {
+      Value::Object(rules)
+    } else {
+      json!({ "matchingRules": rules })
+    };
+    let rules = match matchers_from_json(&rules, &None) {
+      Ok(rules) => rules,
+      Err(err) => {
+        error!("with_matching_rules: Failed to load the matching rules: {}", err);
+        return Ok(false);
+      }
+    };
+
+    interaction.with_interaction(&move |_, _, inner| {
+      if let Some(reqres) = inner.as_v4_http_mut() {
+        reqres.request.matching_rules_mut().merge(&rules);
```

Comment:
This function only allows setting the matching rules for the request. It needs to also allow setting the matching rules for the response.